### PR TITLE
doc/md: add instructions for migration dir linting in ci page

### PR DESCRIPTION
**Pull request by rotemtam**

### CODE REVIEW

File: doc/md/ci.mdx [hunk 1]
```diff
@@ -94,3 +92,166 @@ fi
...
+
+Changes to your project's Ent schema almost always result in a modification
+of your database. If you are using [Versioned Migrations](/docs/versioned-migrations)
+to manage changes to your database schema, we recommend running [migration linting](https://atlasgo.io/versioned/lint)
+as part of your continuous integration flow. This is done for multiple reasons:
+
+* Linting replays your migration directory on a real database to make sure all SQL statements
+  are valid and in the correct order.
+* Directory integrity is enforced - ensuring that history wasn't accidentally changed and
+  that migrations that are planned in parallel are unified to a clean linear history.
```
Comment (masseelch) [inline doc/md/ci.mdx:106]: Link to https://entgo.io/docs/versioned-migrations#atlas-migration-directory-integrity-file

File: doc/md/ci.mdx [hunk 2]
```diff
@@ -94,3 +92,166 @@ fi
...
+  lint:
+    services:
+      # Spin up a mysql:8.0.29 container to be used as the dev-database for analysis.
+      mysql:
+        image: mysql:8.0.29
+        env:
+          MYSQL_ROOT_PASSWORD: pass
+          MYSQL_DATABASE: test
+        ports:
+          - "3307:3306"
```
Comment (masseelch) [inline doc/md/ci.mdx:147]: You are quoting the ports here but not on maria and postgres

File: doc/md/ci.mdx [hunk 3]
```diff
@@ -94,3 +92,166 @@ fi
...
+      - master
+  # Run on PRs where something changed under the `ent/migrate/migrations/` directory.
+  pull_request:
+    paths:
+      - 'ent/migrate/migrations/*'
+jobs:
+  lint:
+    services:
+      # Spin up a mysql:8.0.29 container to be used as the dev-database for analysis.
+      mysql:
```
Comment (masseelch) [inline doc/md/ci.mdx:141]: This service is called `mysql` but maria and postgres are called with their version `postgres10` / `maria107`

File: doc/md/ci.mdx [hunk 4]
```diff
@@ -94,3 +92,166 @@ fi
...
+          --health-retries 10
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3.0.1
+        with:
+          fetch-depth: 0 # Mandatory unless "latest" is set below.
+      - uses: ariga/atlas-action@v0
+        with:
+          dir: path/to/migrations
+          dir-format: golang-migrate # Or: atlas, goose, dbmate
```
Comment (masseelch) [inline doc/md/ci.mdx:206]: We have flyway and liquibase as well.

Also we already depcrecated golang migrate as default. I think we can use `atlas` here already.

File: doc/md/ci.mdx [hunk 5]
```diff
@@ -94,3 +92,167 @@ fi
...
 </Tabs>
+
+## Lint migration files
+
+Changes to your project's Ent schema almost always result in a modification
+of your database. If you are using [Versioned Migrations](/docs/versioned-migrations)
+to manage changes to your database schema, we recommend running [migration linting](https://atlasgo.io/versioned/lint)
+as part of your continuous integration flow. This is done for multiple reasons:
+
+* Linting replays your migration directory on a real database to make sure all SQL statements
```
Comment (a8m) [inline doc/md/ci.mdx:103]: what do you mean by real? let's call it a "container" and link to the Atlas dev-database doc.

File: doc/md/ci.mdx [hunk 6]
```diff
@@ -94,3 +92,167 @@ fi
...
+  are valid and in the correct order.
+* [Migration directory integrity](/docs/versioned-migrations#atlas-migration-directory-integrity-file)
+  is enforced - ensuring that history wasn't accidentally changed and that migrations that are
+  planned in parallel are unified to a clean linear history.
+* Destructive changes are detected notifying you of any potential data loss that may be
+  caused by your migrations way before they reach your production database.
+* Linting detects data-dependant changes that _may_ fail upon deployment and require
+  more careful review from your side.
+
+If you're using GitHub, we recommend using the [Official Atlas Action](https://github.com/ariga/atlas-action)
```
Comment (a8m) [inline doc/md/ci.mdx:113]:
```suggestion
If you're using GitHub, you can use the [Atlas Action](https://github.com/ariga/atlas-action)
```

**Review by a8m — APPROVED** [review]

**Pull request merged**